### ISSUE TITLE
Test Delete user with illegal id checks wrong status

### ISSUE THREAD

**Issue by henrikpalsson** [SAP/scimono#55]
Test Delete user with illegal id and verify Http status code: 400 but according to https://tools.ietf.org/html/rfc7644#section-3.6 it should return 404 Not found

**Comment by karaimin** [comment]
The `ILLEGAL_UUID` used in this test seems to be valid one, which is an issue. However, the purpose of this test is to test for wrong input. It has to test the possibility of `IdValidator` to detect invalid Id format. So, the most appropriate status for wrong input is 400

**Comment by henrikpalsson** [comment]
You are right my misstake. I am now trying to add a CustomInputValidator but I don't even get any call in the isValid functions in IdValidator class. Do I need to do anything to turn it on?

**Comment by karaimin** [comment]
You must override `getCustomConstraintValidators` from `SCIMApplication` class and provide your implementations of `CustomInputValidator`.

**Comment by henrikpalsson** [comment]
I have already done that but it get any call to that function.

**Comment by karaimin** [comment]
Hmm... I am not able to reproduce it. Did you provide correct `CustomInputValidator.Type`. Can you provide me with some code(It can be with stub implementation not the real validator)

**Comment by henrikpalsson** [comment]
public class UuidValidator implements CustomInputValidator {

    public boolean isValid(Object var1){
        if(var1 instanceof String){
            try {
                UUID.fromString((String) var1);
                return true;
            }catch (Exception e){
                return false;
            }
        }else{
            return false;
        }
    }
}

 public Map<CustomInputValidator.Type, CustomInputValidator> getCustomConstraintValidators() {
        Map<CustomInputValidator.Type, CustomInputValidator> validatorMap = new HashMap<>();
        validatorMap.put(CustomInputValidator.Type.USER_ID, new UuidValidator());
        validatorMap.put(CustomInputValidator.Type.GROUP_ID, new UuidValidator());
        validatorMap.put(CustomInputValidator.Type.RESOURCE_ID, new UuidValidator());
        return validatorMap;
    }

**Comment by karaimin** [comment]
> I have already done that but it get any call to that function.

Keep in mind that `IdValidator` class is with application scope. So one instance for whole application is provided by the container. The method  `getCustomConstraintValidators ` only once (when it is first used). Can you check the instance field of `customValidators` of `IdValidator`. How many instances it contains

**Comment by henrikpalsson** [comment]
How can I check that, if I set a debug breakpoint in the functions isValid in class idValidator I don't get trigged, so I have I have hard to check the size of customValidators.

**Comment by karaimin** [comment]
Do you use any Java Bean Validation (JSR-380) implementations in your app. E.g on my test Scimono-server implementation I use:

```
<dependency>
    <groupId>org.glassfish.jersey.ext</groupId>
    <artifactId>jersey-bean-validation</artifactId>
    <version>...</version>
</dependency>
```

**Comment by henrikpalsson** [comment]
No I don't, I can try to add this dependency.

**Comment by karaimin** [comment]
> No I don't, I can try to add this dependency.

It depends on what you use for JAX-RS implementation. I user Jersey so this dependency works for me. If you use other implementation, maybe there are similar handlings

**Comment by henrikpalsson** [comment]
I am using JAX-RS also. Do I need to start it? Just add the dependency didn't work...

**Comment by karaimin** [comment]
Is it possible to show your pom / gradle  dependencies

**Comment by henrikpalsson** [comment]
I am sorry that will be hard, we have build our own dependency files.
Maybe it is something how we start the web application, we start it like this.
CXFNonSpringJaxrsServlet servlet = new CXFNonSpringJaxrsServlet() {
            @Override
            protected void configureSingleton(Object instance) {
                ((SCIMRestApplication) instance).init();
            }
        };

        ServletHolder holder = new ServletHolder(servlet);
        holder.setInitParameter("javax.ws.rs.Application", SCIMRestApplication.class.getName());

**Comment by karaimin** [comment]
You must provide Bean validator implementation someway. As I can see, you use Apache CXF. You shoud find a way to provide Bean validation provider someway. Apache CXF have something `BeanValidationProvider`. And don't know much about the it configuration but I guess It has a way to plug it into provider classes and make it work.
